File: Data Structures & Algorithms/reverse-a-linked-list/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode reverseList(ListNode head) {


        Stack<ListNode> stack = new Stack<ListNode>();
        // create a stack and push the element to the stack 
        while(head != null) {
            stack.push(head);
            head = head.next;
        }


        ListNode dummy = new ListNode(-1);
        head = dummy;

        // pop it up from the stack and form the linkedlist
        while(!stack.isEmpty()) {
            ListNode node = stack.pop();
            ListNode newnode = new ListNode(node.val);

            head.next = newnode;
            head = head.next;
        }


        return dummy.next;
    }
}

File: Data Structures & Algorithms/remove-linked-list-elements/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:

        """
            create a dummy value and dummy.next is head 

            loop and see if the next value is the target 

            curr.next = curr.next.next
        """
        if not head:
            return None

        dummy = ListNode(-1)
        dummy.next = head

        curr = dummy

        while curr.next:
            print(curr.val)
            if curr.next.val == val:
                curr.next = curr.next.next
            else:
                curr = curr.next

        return dummy.next

        